src/app/application.cpp

```diff
@@ -907,20 +907,27 @@ int Application::exec()
             QCoreApplication::exit(EXIT_FAILURE);
         connect(m_webui, &WebUI::fatalError, this, []() { QCoreApplication::exit(EXIT_FAILURE); });
 
-        const Preferences *pref = Preferences::instance();
+        printf("%s", qUtf8Printable(u"\n******** %1 ********\n"_s.arg(tr("Information"))));
 
-        const auto scheme = pref->isWebUiHttpsEnabled() ? u"https"_s : u"http"_s;
-        const auto url = u"%1://localhost:%2\n"_s.arg(scheme, QString::number(pref->getWebUiPort()));
-        const QString mesg = u"\n******** %1 ********\n"_s.arg(tr("Information"))
-                + tr("To control qBittorrent, access the WebUI at: %1").arg(url);
-        printf("%s\n", qUtf8Printable(mesg));
+        if (m_webui->isEnabled())
+        {
+            const auto address = m_webui->hostAddress();
+            const auto url = u"%1://%2:%3"_s.arg(m_webui->isHttps() ? u"https"_s : u"http"_s
```

Comment:
```suggestion
            const QString url = u"%1://%2:%3"_s.arg(m_webui->isHttps() ? u"https"_s : u"http"_s
```
since expression is too complex to easily infer it is a QString.